src/main/scala/com/fulcrumgenomics/sv/tools/SvPileup.scala

```diff
@@ -121,157 +120,166 @@ class SvPileup
-    type BreakpointId = Long
+  type BreakpointId = Long
+
+  /** Value used when no breakpoints are detected. */
+  private val NoBreakpoints: IndexedSeq[BreakpointEvidence] = IndexedSeq.empty
+
+  /** Tracks counts of split reads vs. read pairs supporting a Breakpoint. */
+  private case class BreakpointInfo(id: BreakpointId, var splitRead: Int = 0 , var readPair: Int = 0) {
+    def total: Int = splitRead + readPair
+  }
 
-    private var _nextId: Long   = 0L
-    private val breakpointToId = mutable.HashMap[Breakpoint, BreakpointId]()
-    private val counts         = mutable.HashMap[Breakpoint, Array[Int]]()
+  /** Class that tracks IDs and counts for Breakpoints during discovery. */
+  private class BreakpointTracker extends Iterable[(Breakpoint, BreakpointInfo)] {
+    private var _nextId: Long = 0L
+    private val bpToInfo      = mutable.HashMap[Breakpoint, BreakpointInfo]()
 
     private def nextId: BreakpointId = yieldAndThen(this._nextId) { this._nextId += 1 }
 
     /** Adds a count of one to the evidence for the given breakpoint under the evidence type given.
      * Returns the numerical ID of the breakpoint.
      */
     def count(bp: Breakpoint, ev: EvidenceType): BreakpointId = {
-      val id = this.breakpointToId.getOrElseUpdate(bp, nextId)
-      val ns = this.counts.getOrElseUpdate(bp, new Array[Int](EvidenceType.values.size))
-      ns(EvidenceType.indexOf(ev)) += 1
-      id
+      val info = this.bpToInfo.getOrElseUpdate(bp, BreakpointInfo(nextId))
+
+      ev match {
+        case SplitRead => info.splitRead += 1
+        case ReadPair  => info.readPair  += 1
+      }
+
+      info.id
     }
 
-    override def iterator: Iterator[Breakpoint] = this.breakpoints
+    override def iterator: Iterator[(Breakpoint, BreakpointInfo)] = this.bpToInfo.iterator
 
     /** Returns an iterator over the set of observed breakpoints, ordering is not predictable. */
-    def breakpoints: Iterator[Breakpoint] = breakpointToId.keysIterator
-
-    /** Gets the ID of a breakpoint that has previously been tracked. */
-    def idOf(bp: Breakpoint): BreakpointId = this.breakpointToId(bp)
+    def breakpoints: Iterator[Breakpoint] = bpToInfo.keysIterator
 
     /** Gets the evidence counts for a given breakpoint in the same order as EvidenceType.values. */
-    def countsFor(bp: Breakpoint): Array[Int] = this.counts(bp)
+    def apply(bp: Breakpoint): BreakpointInfo = this.bpToInfo(bp)
+  }
 
-    /** Gets the evidence counts for a given breakpoint zipped with the evidence types. */
-    def evidenceFor(bp: Breakpoint): IndexedSeq[(EvidenceType, Int)] = {
-      EvidenceType.values.zip(this.counts(bp))
-    }
+  /**
+   * Performs filtering on a Template to remove primary records that are unmapped and then remove
+   * supplementary records if there is no matching primary record retained.  If neither primary
+   * record is retained, returns None, else returns Some(Template).
+   */
+  def filterTemplate(t: Template,
```

Comment:
This could definitely be optimized a bit in order to not create copies of things when not needed, but that would make it more complicated and I think this is fine for now.